### DISCUSSION TITLE
Performance and Stability

### DISCUSSION THREAD

**Discussion by EliteMasterEric** [EliteMasterEric/Teyvat.moe#261] [Ideas]
This is boring but necessary work.

- [ ] Develop Jest test suite to ensure proper functionality of all components.
- [ ] Develop scripts to convert and optimize images as necessary.
- [x] Integrate webp images on supported browsers.

# Performance tracking tools
- https://observatory.mozilla.org/analyze/
- Chrome Lighthouse
- https://www.ssllabs.com/ssltest/
- https://web.dev/vitals/
- https://www.webpagetest.org/
- https://www.thinkwithgoogle.com/feature/testmysite
- https://developers.google.com/speed/pagespeed/insights/
- https://www.uptrends.com/tools/website-speed-test
- https://tools.pingdom.com/
- https://www.whatismyip.org/website-reviewer/

**Comment by sgtoutlaw** [comment]
Did you at some point compress the images to the map?   It looks washed out now.  

This wouldn't be a problem but I noticed it when I was trying to match up the map with my game map for marking.

Maybe it is just me.  That or it was a necessary sacrifice, which I can accept.  

I just wanted to ask and this looked like maybe a better place to ask then writing it as a bug, since it could be as intended.

**Reply by EliteMasterEric** [reply]
I've done some optimizations that shouldn't cause a noticeable difference in image quality. I can double check.

**Comment by sgtoutlaw** [comment]
![firefox_Al27hEkSV2](https://user-images.githubusercontent.com/35310360/104226759-4c3cd180-5416-11eb-8ecd-0f9a1e27ccf8.png)
site

![GenshinImpact_58MIWXSBGE](https://user-images.githubusercontent.com/35310360/104226802-5bbc1a80-5416-11eb-9d88-e769cdf7477d.png)
game

This is where I was marking chests earlier, and ususally the (almost wavy?) lines in the map would help me. I realized had nothing to work with anymore.

**Reply by EliteMasterEric** [reply]
I believe I have narrowed down the root cause of this issue. There appears to be a loss in quality when converting the image to a WEBP format, which reduces the file size significantly (more than 90%) but also damages image fidelity.

I will look into this.
